aws_lambda_powertools/tracing/tracer.py

```diff
@@ -625,8 +629,8 @@ def _add_response_as_metadata(
         self,
         method_name: str = None,
         data: Any = None,
-        subsegment: aws_xray_sdk.core.models.subsegment = None,
-        capture_response: Optional[bool] = None,
+        subsegment: BaseSegment = None,
```

Comment:
__Question__: could this break anything since aws_xray_sdk.core.models.subsegment does not inherit BaseSegment, but fulfill its methods properly?

Reply:
not in this case no, it's primarily for type hinting since adding `aws_xray_sdk.core.models.subsegment` triggers a full load throwing the lazy import effectiveness out of the window. This custom type/ABC is an entry level change to something bigger - Bring Your Own Tracer Provider in 2.0